src/main/java/com/azure/cosmos/examples/changefeed/SampleChangeFeedProcessor.java

```diff
@@ -0,0 +1,269 @@
+ */
+public class SampleChangeFeedProcessor {
+
+    public static int WAIT_FOR_WORK = 60000;
+    public static final String DATABASE_NAME = "db_" + RandomStringUtils.randomAlphabetic(7);
+    public static final String COLLECTION_NAME = "coll_" + RandomStringUtils.randomAlphabetic(7);
+    private static final ObjectMapper OBJECT_MAPPER = Utils.getSimpleObjectMapper();
+    protected static Logger logger = LoggerFactory.getLogger(SampleChangeFeedProcessor.class.getSimpleName());
+
+
+    private static ChangeFeedProcessor changeFeedProcessorInstance;
+    private static boolean isWorkCompleted = false;
+
+    public static void main(String[] args) {
+        logger.info("BEGIN Sample");
+
+        try {
+
+            //Summary of the next four commands:
+            //-Create an asynchronous Azure Cosmos DB client and database so that we can issue async requests to the DB
+            //-Create a "feed container" and a "lease container" in the DB
+            logger.info("-->CREATE DocumentClient");
+            CosmosAsyncClient client = getCosmosClient();
+
+            logger.info("-->CREATE sample's database: " + DATABASE_NAME);
+            CosmosAsyncDatabase cosmosDatabase = createNewDatabase(client, DATABASE_NAME);
+
+            logger.info("-->CREATE container for documents: " + COLLECTION_NAME);
+            CosmosAsyncContainer feedContainer = createNewCollection(client, DATABASE_NAME, COLLECTION_NAME);
+
+            logger.info("-->CREATE container for lease: " + COLLECTION_NAME + "-leases");
+            CosmosAsyncContainer leaseContainer = createNewLeaseCollection(client, DATABASE_NAME, COLLECTION_NAME + "-leases");
+
+            //Model of a worker thread or application which leases access to monitor one or more feed container
+            //partitions via the Change Feed. In a real-world application you might deploy this code in an Azure function.
+            //The next line causes the worker to create and start an instance of the Change Feed Processor. See the implementation of getChangeFeedProcessor() for guidance
+            //on creating a handler for Change Feed events. In this stream, we also trigger the insertion of 10 documents on a separate
+            //thread.
+            logger.info("-->START Change Feed Processor on worker (handles changes asynchronously)");
+            changeFeedProcessorInstance = getChangeFeedProcessor("SampleHost_1", feedContainer, leaseContainer);
+            changeFeedProcessorInstance.start()
+                    .subscribeOn(Schedulers.elastic())
+                    .doOnSuccess(aVoid -> {
+                        //pass
+                    })
+                    .subscribe();
+
+            //These two lines model an application which is inserting ten documents into the feed container
+            logger.info("-->START application that inserts documents into feed container");
+            createNewDocumentsCustomPOJO(feedContainer, 10, Duration.ofSeconds(3));
```

Comment:
Instead of using a bare bone CustomPOJO, it might be worth using one of the other POJO classes such a Family for instance. It will help make it more clear how properties will map and be accessed.